documentation/asciidoc/topics/proc_enabling_client_cert_authn.adoc

```diff
@@ -0,0 +1,36 @@
+[id='enabling-client-certificate-authentication_{context}']
+= Enabling client certificate authentication
+
+[role="_abstract"]
+{brandname} clusters do not use client certificate authentication by default.
+Complete the following procedure so that clients must present valid certificates before they can connect to {brandname} endpoints.
+
+You can configure {brandname} to verify client identities from certificates in a trust store in two ways:
+
+`Validate`:: Requires a trust store that contains only the signing certificate, which is typically a Certificate Authority (CA). Any client that presents a signed certificate can connect to {brandname}.
+`Authenticate`:: Requires a trust store that contains all client certificates in addition to the signing certificate. Only clients that present a signed certificate that is present in the trust store can connect to {brandname}.
+
+.Procedure
+
+. Set either `Validate` or `Authenticate` as the value for the `spec.security.endpointEncryption.clientCert` field in your `Infinispan` CR.
++
+[NOTE]
+====
+The default value is `None`.
+====
++
+. Optionally specify a name for the secret that contains the client trust store with the `spec.security.endpointEncryption.clientCertSecretName` field.
```

Comment:
This isn't optional. The user must always provide `clientCertSecretName`, otherwise there are 1) no certificates to include in a truststore created by the operator or 2) No existing truststore to use.

Reply:
@ryanemerson Right after this we state: "By default Infinispan Operator adds the truststore to a secret named `example-infinispan-client-cert-secret`.

So users should specify the default secret name if not using a custom secret?

Reply:
> "By default Infinispan Operator adds the truststore to a secret named `example-infinispan-client-cert-secret`.

This is wrong. A trustsore secret is never generated by the operator, it has to be provided by the user as this is the only way to get the truststore or certs to create the truststore. 

I'm guessing I failed to update my docs primer with this detail. Originally the idea was that a truststore could be generated using the Openshift service CA, but this doesn't make sense and was scrapped.

Reply:
I'm suffering from brain fog myself (and the internet is acting weird today). I've updated with some changes, hopefully they make sense.